persistence/pre_persistence/genesis.go

```diff
@@ -25,18 +27,27 @@ func NewGenesisState(numOfValidators, numOfApplications, numOfFisherman, numOfSe
 	vm := testingState.ValidatorMap
 	// generate `mocked` keys for each actor
 	// specifically, use the test state json file to create the validator keys
-	validatorKeys = make([]crypto.PrivateKey, len(vm))
+	if cfg.IsTesting {
+		validatorKeys = make([]crypto.PrivateKey, numOfValidators)
+	} else {
+		validatorKeys = make([]crypto.PrivateKey, len(vm))
+	}
```

Comment:
_On a lighter note_:

Imagine Go had ternaries 😢 
It's too complicated they said, too unreadable.